demos/browser/app/messagingSession/messagingSession.ts

```diff
@@ -66,6 +67,23 @@ export class DemoMessagingSessionApp implements MessagingSessionObserver {
         console.error(`Failed to retrieve messaging session endpoint: ${error.message}`);
       }
     });
+    document.getElementById('connectWithPrefetch').addEventListener('click', async () => {
+      try {
+        const response = await this.fetchCredentials();
```

Comment:
You should be able to reuse the code above. No need to duplicate all of these code.